client/src/client.rs

```diff
@@ -370,14 +370,10 @@ where
     /// its own thread.
     pub fn run_with<P>(mut self, listen: Vec<net::SocketAddr>, protocol: P) -> Result<(), Error>
     where
-        P: nakamoto_net::Service<Event = fsm::Event, Command = Command>,
+        P: nakamoto_net::PeerService<Notification = fsm::Event, Command = Command>,
```

Comment:
```suggestion
        P: nakamoto_net::PeerService<Notification = fsm::Event, Command = Command>,
```

Mh, also in favor to keep `Event`

Reply:
Why I changed the name is because it is not clear where event is originated and goes to, while notification clearly goes from the protocol to the user. But no strong opinion, I'm fine with reverting it.